### PR TITLE
module 9 Spring MVC & JSP

### PR DESCRIPTION
**Pull request by alextimonov**

### CODE REVIEW

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcDishDAO.java [hunk 1]
```diff
@@ -61,6 +67,20 @@ public void delete(String name) {
         template.update(sql, name);
     }
 
+    @Override
+    public List<Ingredient> defineDishIngredients(Dish dish) {
+        List<Ingredient> result = new ArrayList<>();
+        String sql = "SELECT Ingredient.* \n" +
+                "FROM (Ingredient INNER JOIN Ingredient_to_dish ON Ingredient_to_dish.ingredient_id = Ingredient.id) \n" +
+                "WHERE Ingredient_to_dish.dish_id = ?";
+        List<Map<String, Object>> mapList = template.queryForList(sql, dish.getId());
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcDishDAO.java:76]: Use Java 8 Stream to transform collections

``` java
mapList.stream()
   .map(row -> jdbcStorageDAO.getIngredientFromMap(row))
   .collect(Collectors.toList);
```

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcEmployeeDAO.java [hunk 2]
```diff
@@ -79,6 +80,20 @@ public void delete(String name, String surname) {
             template.update(sql, name, surname);
     }
 
+    @Override
+    public List<Waiter> getWaiters() {
+        List<Waiter> result = new ArrayList<>();
+        String sql = "SELECT Employee.id, Employee.surname, Employee.name, Jobs.position, Employee.birthday, Employee.salary\n" +
+                "FROM Employee INNER JOIN Jobs ON Employee.position_id = Jobs.id WHERE Employee.dtype = 'WAITER'";
+        List<Map<String, Object>> mapList = template.queryForList(sql);
+        for (Map<String, Object> row : mapList) {
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcEmployeeDAO.java]: The same about streams

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcDishDAO.java (hunk 1 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcDishDAO.java:76]: fixed in all Jdbc..DAO classes

File: src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcEmployeeDAO.java (hunk 2 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcEmployeeDAO.java]: fixed in all Jdbc..DAO classes

File: src/main/java/ua/goit/timonov/enterprise/module_9/web/DishWebController.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Map;
+
+/**
+ * Controller for mapping requests to Dish pages
+ */
+
+@Controller
+public class DishWebController {
+
+    private DishService dishService;
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_9/web/DishWebController.java:26]: For me Constructor injection is preferrable.
Not sure if Spring supports it

File: src/main/java/ua/goit/timonov/enterprise/module_9/web/MainController.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @RequestMapping(value = "/main", method = RequestMethod.GET)
+    public String getMainPage() {
+        return "main";
+    }
+
+    @RequestMapping(value = "/contacts", method = RequestMethod.GET)
+    public String getPageContacts() {
+        return "contacts";
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_9/web/MainController.java]: Use Constants here

File: src/main/webapp/WEB-INF/view/jsp/menus.jsp [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    <title>My Restaurant|Menus</title>
+</head>
+<body>
+
+<h2>Our menus</h2>
+<c:forEach var="menu" items="${menus}">
+    <h3>Menu ${menu.name}</h3>
+    <table border="1">
+        <tr>
+            <th> ID </th>
```
Comment (schaffe) [inline src/main/webapp/WEB-INF/view/jsp/menus.jsp]: remove whitespaces in xml/html/whatever

**Review by schaffe — CHANGES_REQUESTED** [review]

File: src/main/java/ua/goit/timonov/enterprise/module_9/web/DishWebController.java (hunk 3 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_9/web/DishWebController.java:26]: fixed, Spring supports it

File: src/main/java/ua/goit/timonov/enterprise/module_9/web/MainController.java (hunk 4 above)
Comment (alextimonov) [reply src/main/java/ua/goit/timonov/enterprise/module_9/web/MainController.java]: fixed

File: src/main/webapp/WEB-INF/view/jsp/menus.jsp (hunk 5 above)
Comment (alextimonov) [reply src/main/webapp/WEB-INF/view/jsp/menus.jsp]: fixed

File: src/main/java/ua/goit/timonov/enterprise/module_9/web/DishWebController.java [hunk 6]
```diff
@@ -20,32 +19,35 @@
...
 
+    public static final String DISHES_PAGE = "dishes";
+    public static final String DISH_PAGE = "dish";
+    public static final String NOT_FOUND_PAGE = "itemNotFound";
+    public static final String FIND_BY_NAME_PAGE = "findDishByName";
     private DishService dishService;
 
     @Autowired
-    public void setDishService(DishService dishService) {
+    public DishWebController(DishService dishService) {
```
Comment (schaffe) [inline src/main/java/ua/goit/timonov/enterprise/module_9/web/DishWebController.java:29]: in future: if using constructor injection mark injected fields as `final`

**Comment by schaffe** [comment]
to be merged

**Pull request merged**